main/src/main/MainScopts.scala

```diff
@@ -25,7 +25,5 @@ object Tasks{
 class EvaluatorScopt[T]()
   extends scopt.Read[mill.eval.Evaluator]{
   def arity = 0
-  def reads = s => {
-    Evaluator.currentEvaluator.get.asInstanceOf[mill.eval.Evaluator]
-  }
+  def reads = _ => Evaluator.currentEvaluator.get
```

Comment:
It would be consequent to also make the return code explicit.

Reply:
If we want to make it explicit we should do `Evaluator.currentEvaluator.get: mill.eval.Evaluator`
Casting is bad as you shortcut the type checker.